src/validation/data/service.rs

```diff
@@ -0,0 +1,9 @@
+use super::*;
+
+/// Service definition.
+#[derive(Serialize, Deserialize, Default, Clone, Debug, PartialEq, Eq, Diff)]
+#[diff(attr(
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+))]
+pub struct Service {}
```

Comment:
If this is pub, can we type this correctly?

Reply:
Please remove the types that are not used / or implement them

Reply:
I would suggest that you track a list of types to be implemented in the module documentation of the data mod so we know what is still missing 😊